sequencer/api/migrations/V38__upgrade_certificate.sql

```diff
@@ -0,0 +1,6 @@
+CREATE TABLE upgrade_certificate (
+     -- The ID is always set to 0. Setting it explicitly allows us to enforce with every insert or
+    -- update that there is only a single entry in this table: the latest known state.
+    id INT PRIMARY KEY,
```

Comment:
@imabdulbasit I think we could enforce at the DB level that there is at most 1 upgrade certificate. Should we do that?

Reply:
I think the only way to do it is to add a trigger which is fine but I don't know how frequently we are storing the upgrade certificate as the trigger will do a query on the table and then insert

Reply:
It should be accessed very rarely. We (should) write exactly once per upgrade, and read only on node restart.

Reply:
@imabdulbasit What about something like what's proposed here: https://stackoverflow.com/a/25393923 ?

Reply:
yeah, this should work fine. cc @lukeiannucci

Reply:
made changes, can you verify if this looks good @imabdulbasit / @sveitser . CI seems good also added a test